src/internal/Confirm.tsx

```diff
@@ -0,0 +1,165 @@
+import * as React from 'react';
+import { Drop, DropProps } from 'grommet';
+import styled from 'styled-components';
+import { px } from '../utils';
+import { Box, Button } from '..';
+import TooltipArrow, { TooltipArrowProps } from './TooltipArrow';
+
+const StyledDrop = styled(Drop)`
+	border: 0;
+`;
+
+const StyledDropInner = styled.div`
+	margin: 5px;
+	background: white;
+	box-shadow: ${(props) => '1px 1px 5px' + props.theme.colors.gray.light};
+	border-radius: ${(props) => px(props.theme.radius)};
+	border: ${(props) => '1px solid ' + props.theme.colors.gray.main};
+`;
+
+const getDropAlign = (
+	placement: TooltipArrowProps['placement'],
+): DropProps['align'] => {
+	switch (placement) {
+		case 'bottom':
+			return {
+				top: 'bottom',
+			};
+		case 'top':
+			return {
+				bottom: 'top',
+			};
+		case 'left':
+			return {
+				right: 'left',
+			};
+		case 'right':
+			return {
+				left: 'right',
+			};
+	}
+};
+
+export interface ConfirmOptions {
+	text: string;
+	placement?: TooltipArrowProps['placement'];
+}
+
+interface ConfirmProps extends DropProps, ConfirmOptions {
+	targetRef: React.RefObject<HTMLElement>;
+	onOK: () => any;
+	onCancel: () => any;
+}
+
+const Confirm = ({
```

Comment:
This might be annoying to hear at this stage but this feels like it should be two components:
1. A `Popover` component that just handles the placement and styling of the absolutely positioned element (including triangle) and doesn't care what is inside it.
2. A `Confirm` component  that uses a `Popover` internally and handles the text and cancel & confirm buttons.

There are definitely other uses for the `Popover` so I would promote that to be an exported component. (I'd use it in Jellyfish for sure!)